pkgs/extension_discovery/lib/extension_discovery.dart

```diff
@@ -156,6 +169,22 @@ Future<List<Extension>> findExtensions(
   );
 }
 
+Uri? _findPackageConfig(Uri packageDir) {
+  if (!packageDir.path.endsWith('/')) {
+    packageDir = packageDir.replace(path: '${packageDir.path}/');
+  }
+  while (true) {
+    final packageConfigCandidate =
+        packageDir.resolve('.dart_tool/package_config.json');
+    if (File.fromUri(packageConfigCandidate).existsSync()) {
+      return packageConfigCandidate;
+    }
```

Comment:
```suggestion
    try {
      if (File.fromUri(packageConfigCandidate).existsSync()) {
        return packageConfigCandidate;
      }
    } on IOException {
      return null; // if we get a permission error, etc, we return null
    }
```

Reply:
I don't mind - it is not documented if `existsSync` can throw.

Done